src/v/storage/tests/compaction_e2e_test.cc

```diff
@@ -224,6 +301,419 @@ TEST_P(CompactionFixtureParamTest, TestDedupeOnePass) {
 INSTANTIATE_TEST_SUITE_P(
   DuplicatesPerKey, CompactionFixtureParamTest, ::testing::Values(1, 10, 100));
 
+TEST_F(CompactionFixtureTest, TestTombstones) {
+    auto duplicates_per_key = 2;
+    auto total_records = 10;
+    auto cardinality = total_records / duplicates_per_key;
+    auto num_segments = 5;
+    size_t record_batches_per_segment = total_records / num_segments;
+
+    generate_data(num_segments, cardinality, record_batches_per_segment).get();
+    // Generate a tombstone record for "key0".
+    generate_tombstones(1, 1, 1).get();
+
+    auto tombstone_retention_ms = 1000ms;
+
+    // Helper function to perform sliding window compaction.
+    auto do_sliding_window_compact = [tombstone_retention_ms](auto& log) {
```

Comment:
Given it's used below too, think this is worth moving into a method of CompactionFixtureTest?